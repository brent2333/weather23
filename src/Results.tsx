import { Fragment } from "react";
import { CurrentWeatherResponse, SharedLocationObject } from "./APIResponseTypes";
import { CloudIcon, SunIcon } from "@heroicons/react/24/solid";

const Results = ({ current }: CurrentWeatherResponse, loc: SharedLocationObject) => {
  return (
    <Fragment>
      <h2 className="mx-6 text-white w-full basis-full">Current Weather</h2>
      <div className="lg:grid lg:grid-flow-col sm:flex md:flex mx-auto my-0 max-w-screen-lg">
        <div className="shadow-md rounded m-4 text-3xl font-bold center-flex p-8 bg-gray-300">
          {current?.condition.text}&nbsp;
          {current?.condition.icon ? (
          <img src={`https:${current?.condition.icon}`} alt={current?.condition.icon} data-testid="condimg" />
          ) : null}
        </div>
        {current?.temp_f ? (
          <div className="shadow-md rounded m-4 text-3xl font-extrabold center-flex p-8 bg-gray-300">
            {current?.temp_f} <span>&#176;</span>F
          </div>
        ) : null}
        {current?.temp_c ? (
          <div className="shadow-md rounded  m-4 text-3xl font-extrabold center-flex p-8 bg-gray-300">
            {current?.temp_c} <span>&#176;</span>C
          </div>
        ) : null}
        {current?.feelslike_f ? (
          <div className="shadow-md rounded  m-4 text-3xl pb-8 pt-2 px-6 bg-gray-300">
            <span className="text-base">Feels like</span>
            <div className="font-extrabold">{current?.feelslike_f} <span>&#176;</span>F</div>
            <div className="font-extrabold">{current?.feelslike_c} <span>&#176;</span>C</div>

          </div>
        ) : null}
      </div>
    </Fragment>
  );
};

export default Results;
