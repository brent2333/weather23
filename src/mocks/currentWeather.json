{
    "last_updated_epoch": 1673291700,
    "last_updated": "2023-01-09 19:15",
    "temp_c": 7.0,
    "temp_f": 44.6,
    "is_day": 0,
    "condition": {
        "text": "Clear",
        "icon": "//cdn.weatherapi.com/weather/64x64/night/113.png",
        "code": 1000
    },
    "wind_mph": 16.1,
    "wind_kph": 25.9,
    "wind_degree": 270,
    "wind_dir": "W",
    "pressure_mb": 1009.0,
    "pressure_in": 29.8,
    "precip_mm": 0.0,
    "precip_in": 0.0,
    "humidity": 70,
    "cloud": 0,
    "feelslike_c": 3.6,
    "feelslike_f": 38.5,
    "vis_km": 10.0,
    "vis_miles": 6.0,
    "uv": 1.0,
    "gust_mph": 14.1,
    "gust_kph": 22.7
}