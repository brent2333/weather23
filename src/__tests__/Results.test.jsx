import { expect, test } from "vitest";
import { render } from "@testing-library/react";
import { StaticRouter } from "react-router-dom/server";
import Results from "../Results";
import currentWeather from "../mocks/currentWeather.json";

test("renders the correct image for current condition", async () => {
    const results = render(
        <Results current={currentWeather} />
    );
    const condImg = await results.findByTestId("condimg");
    expect(condImg.src).toEqual("https://cdn.weatherapi.com/weather/64x64/night/113.png");
    results.unmount();
  });

  test("renders correctly with weather data", () => {
  const { asFragment } = render(
    <StaticRouter>
      <Results current={currentWeather} />
    </StaticRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
