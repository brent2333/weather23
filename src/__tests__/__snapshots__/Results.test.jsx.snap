// Vitest Snapshot v1

exports[`renders correctly with weather data 1`] = `
<DocumentFragment>
  <h2
    class="mx-6 text-white w-full basis-full"
  >
    Current Weather
  </h2>
  <div
    class="lg:grid lg:grid-flow-col sm:flex md:flex mx-auto my-0 max-w-screen-lg"
  >
    <div
      class="shadow-md rounded m-4 text-3xl font-bold center-flex p-8 bg-gray-300"
    >
      Clear 
      <img
        alt="//cdn.weatherapi.com/weather/64x64/night/113.png"
        data-testid="condimg"
        src="https://cdn.weatherapi.com/weather/64x64/night/113.png"
      />
    </div>
    <div
      class="shadow-md rounded m-4 text-3xl font-extrabold center-flex p-8 bg-gray-300"
    >
      44.6 
      <span>
        °
      </span>
      F
    </div>
    <div
      class="shadow-md rounded  m-4 text-3xl font-extrabold center-flex p-8 bg-gray-300"
    >
      7 
      <span>
        °
      </span>
      C
    </div>
  </div>
</DocumentFragment>
`;
